### PR TITLE
Implement ChecksumCalculator Interface

### PR DESCRIPTION
**Pull request by damijank**
If an application is handling videos, the method to download the file locally and calculate the MD5 hash is not feasible. Azure calculates the MD5 hash, this PR implements fetching MD5 info from Azure.

### CODE REVIEW

**Comment by dkarlovi** [comment]
Closes #590.

**Comment by nicolasmure** [comment]
Hello @damijank :)

Thank you for your contribution, do you mind to add a test covering your changes ?

Thanks ;)

File: tests/Gaufrette/Functional/Adapter/AzureMultiContainerBlobStorageTest.php [hunk 1]
```diff
@@ -108,6 +108,21 @@ public function shouldGetSize()
...
+     */
+    public function shouldGetMd5Hash()
+    {
+        $path = $this->createUniqueContainerName('container') . '/foo';
+
+        $content = 'Some content';
+        $this->filesystem->write($path, $content);
+        $checksumBase64 = $this->filesystem->checksum($path);
+
+        $this->assertEquals(\md5($content), \bin2hex(\base64_decode($checksumBase64, true)));
```
Comment (nicolasmure) [inline tests/Gaufrette/Functional/Adapter/AzureMultiContainerBlobStorageTest.php]: I'm not sure to get your test here. Why to call again `bin2hex` and `base64_decode` ?
Shouldn't the test be
```php
$checksum = $this->filesystem->checksum($path);
$this->assertEquals(\md5($content), $checksum);
```
?

Comment (damijank) [reply tests/Gaufrette/Functional/Adapter/AzureMultiContainerBlobStorageTest.php]: Yup, you are right. It was too quick with this. Sending fix right away.

**Comment by nicolasmure** [comment]
Thank you @damijank !

**Pull request merged**